demo/regression/machine.conf

```diff
@@ -1,30 +1,28 @@
 # General Parameters, see comment for each definition
 # choose the tree booster, can also change to gblinear
 booster = gbtree
-# this is the only difference with classification, use reg:squarederror to do linear classification
+# this is the only difference with classification, use reg:linear to do linear classification
```

Comment:
Can you remove `demo/regression/machine.conf` from this pull request? It's not relevant to the CMake refactor.

Reply:
@hcho3 It's used in a test for CLI.  But I can split it into another PR.